File: push/src/main/java/com/nihility/service/XMPushServiceListener.java
package com.nihility.service;

import android.content.Intent;

public interface XMPushServiceListener {
    default void created() {
    }

    default void destroy() {
    }

    default void start(Intent intent) {
    }

    enum ConnectionStatus {
        connecting,
        connected,
        disconnected,
    }

    default void connectionStatusChanged(ConnectionStatus connectionStatus) {
    }
}

File: mipush_hook/src/main/java/com/magisk317/service/XMPushServiceListener.kt
package com.magisk317.service

import android.content.Intent

interface XMPushServiceListener {
    fun created() {}
    fun destroy() {}
    fun start(intent: Intent) {}

    enum class ConnectionStatus {
        connecting,
        connected,
        disconnected;

        companion object {
            @JvmStatic
            fun of(i: Int): ConnectionStatus {
                return values()[i]
            }
        }
    }

    fun connectionStatusChanged(connectionStatus: ConnectionStatus) {}
}

typealias ConnectionStatus = XMPushServiceListener.ConnectionStatus
